src/create-action-creator-factory.ts

```diff
@@ -0,0 +1,45 @@
+import { ActionCreator } from './create-action-creator'
+import { Action } from './create-action'
+
+
+type ActionCreatorFactory = (type: string) => ActionCreator<string>;
+
+interface ActionCreatorFactoryMapper {
+  [key: string]: ActionCreatorFactory;
+}
+
+type ActionTypesMap<TMapper extends ActionCreatorFactoryMapper> = {
+  [TKey in keyof TMapper]: string;
+};
+
+type ArgumentTypes<T> = T extends (...args: infer U) => any ? U : never;
+type ReplaceCreatorType<TCreator, TNewType extends string> = TCreator extends (
```

Comment:
I think that we can lose some type information here.
I am really not sure if this is the correct way to do it.

Reply:
Approach changed